### PR TITLE
MCR-1771 add mcr counting dnburn generator

### PR DESCRIPTION
**Pull request by sebhofmann**

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on d8d804a9daa73209ea797e591375d5b2056e6ccb! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/2519-pull_request-466

**Review by Mewel — APPROVED** [review]

File: mycore-pi/src/main/java/org/mycore/pi/urn/MCRCountingDNBURNGenerator.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package org.mycore.pi.urn;
```
Comment (kkrebs) [inline mycore-pi/src/main/java/org/mycore/pi/urn/MCRCountingDNBURNGenerator.java:19]: there should be license information in every mycore class

File: mycore-pi/src/main/java/org/mycore/pi/urn/MCRDNBURNGenerator.java [hunk 2]
```diff
@@ -34,15 +34,16 @@ public MCRDNBURNGenerator(String generatorID) {
         super(generatorID);
     }
 
-    protected abstract String buildNISS();
+    protected abstract String buildNISS(MCRObjectID mcrID, String additional);
 
-    public MCRDNBURN generate(@NotNull String namespace, String additional) throws MCRPersistentIdentifierException {
+    public MCRDNBURN generate(@NotNull String namespace, MCRObjectID mcrID, String additional)
```
Comment (kkrebs) [inline mycore-pi/src/main/java/org/mycore/pi/urn/MCRDNBURNGenerator.java]: public methods should have javadoc

**Review by kkrebs — CHANGES_REQUESTED** [review]
see inline comments

**Pull request merged**